src/lib.rs

```diff
@@ -754,6 +767,24 @@ impl Capture<Dead> {
             .map(|h| Capture::new(h))
             .ok_or(InsufficientMemory)
     }
+
+    /// Compiles a pcap filter expression to a bpf program, which is just
+    /// a list of bpf instructions
+    pub fn compile_filter(&mut self, program: &str) -> Result<Vec<bpf_insn>, Error> {
```

Comment:
We could theoretically also implement this for other capture types, but I'm not sure of the utility.